bad oop with JvmtiVTMSTransitionDisabler

alexmenkov opened the pull request:
Bad oop can happen after JvmtiVTMSTransitionDisabler is installed (at java_lang_VirtualThread::carrier_thread(java_thread) call).
Fixed by converting java_thread to Handle

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - Committer)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/loom.git pull/217/head:pull/217` \
`$ git checkout pull/217`

Update a local copy of the PR: \
`$ git checkout pull/217` \
`$ git pull https://git.openjdk.org/loom.git pull/217/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 217`

View PR using the GUI difftool: \
`$ git pr show -t 217`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/loom/pull/217.diff">https://git.openjdk.org/loom/pull/217.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/loom/pull/217#issuecomment-2819719082)
</details>

AlanBateman commented:
Thanks for jumping on this, looks right. There is also an issue with GetThreadSnapshotClosure._name and ThreadSnaphot is temporarily changed to drop the name parameter, do you want to look at this at the same time?

alexmenkov commented:
> Thanks for jumping on this, looks right. There is also an issue with GetThreadSnapshotClosure._name and ThreadSnaphot is temporarily changed to drop the name parameter, do you want to look at this at the same time?

Fixed.
Also fixed "bad oop" when on return from java_lang_Thread::get_thread_snapshot (by destroying JvmtiVTMSTransitionDisabler earlier)

coleenp commented:
Can you run this code with -XX:+CheckUnhandledOops and see if there are more?

AlanBateman commented:
> Can you run this code with -XX:+CheckUnhandledOops and see if there are more?

It was a test run with -XX:+CheckUnhandledOops that caught these issues, the tests are now passing with the changes here, so I think it allows the prototyping work continue.

alexmenkov commented:
/integrate

AlanBateman commented:
/sponsor
